dkpro-core-io-nyt-asl/src/main/java/de/tudarmstadt/ukp/dkpro/core/io/nyt/NYTCollectionReader.java

```diff
@@ -0,0 +1,170 @@
+package de.tudarmstadt.ukp.dkpro.core.io.nyt;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.uima.UimaContext;
+import org.apache.uima.cas.CAS;
+import org.apache.uima.cas.CASException;
+import org.apache.uima.collection.CollectionException;
+import org.apache.uima.fit.component.CasCollectionReader_ImplBase;
+import org.apache.uima.fit.descriptor.ConfigurationParameter;
+import org.apache.uima.jcas.JCas;
+import org.apache.uima.jcas.cas.StringArray;
+import org.apache.uima.resource.ResourceInitializationException;
+import org.apache.uima.util.Progress;
+import org.apache.uima.util.ProgressImpl;
+
+import com.nytlabs.corpus.NYTCorpusDocument;
+
+import de.tudarmstadt.ukp.argumentext.demonstrator.metadata.NYTArticleMetaData;
+import de.tudarmstadt.ukp.dkpro.core.api.metadata.type.DocumentMetaData;
+
+public class NYTCollectionReader extends CasCollectionReader_ImplBase {
```

Comment:
Readers should inherit from JCasResourceCollectionReader_ImplBase which already includes support for archives, recursively traversing folder hierarchies, etc. Mind that TGZ archives are not supported though - only ZIP archives.